ext/rubymain.cpp

```diff
@@ -1036,7 +1037,12 @@ t_unwatch_filename
 
 static VALUE t_unwatch_filename (VALUE self UNUSED, VALUE sig)
 {
-	evma_unwatch_filename(NUM2BSIG (sig));
+	try {
+		evma_unwatch_filename(NUM2BSIG (sig));
+	} catch (std::runtime_error e) {
+		rb_raise (EM_eInvalidSignature, "%s", e.what());
+	}
```

Comment:
I agree with this regardless of the reason for the invalid signature; raising a C++ exception into Ruby space that kills the application isn't helpful to anybody.